docs/decisions/2022-07-pet-test-fixtures.md

```diff
@@ -0,0 +1,73 @@
+# Allow "pet" test fixtures
+
+* **Status:** approved
+* **Last Updated:** 2022-07
+
+## Context & Problem Statement
+
+Emblem evolved to use "pet" test fixtures in end-to-end (E2E) test runs. These include resources such as Google Cloud projects, Cloud Storage buckets, container images in Artifact Registry, and Cloud Run services.
+
+We decided to investigate whether we should migrate our fixtures to a "cattle"-based model.
+
+## Priorities
+
+Ideally, our test fixtures would be designed such that:
+
+* Test infrastructure is quick to set up.
+* Test infrastructure is documented (e.g. via Terraform files).
+* Tests themselves are quick to write.
+* Both tests and their infrastructure require minimal maintenance.
+* Tests are runnable by multiple [teams of] developers in parallel.
+* Tests cover as much of the application surface as possible.
+* Tests allow developers and contributors to quickly identify root causes of bugs.
```

Comment:
Another priority/principle to consider: Tests have minimal reliance on state. Minimal in our case is not "none" or close to it: it feels like we are very close to the maximum amount of state we can manage.

Reply:
Good call - added.